qualtran/bloqs/basic_gates/su2_rotation.py

```diff
@@ -0,0 +1,107 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from functools import cached_property
+from typing import Dict, Tuple, TYPE_CHECKING, Union
+
+import numpy as np
+from attrs import frozen
+from numpy.typing import NDArray
+
+from qualtran import bloq_example, BloqDocSpec, GateWithRegisters, Signature
+from qualtran.bloqs.basic_gates import Ry, ZPowGate
+from qualtran.drawing import TextBox
+
+if TYPE_CHECKING:
+    import cirq
+
+    from qualtran import BloqBuilder, Soquet, SoquetT
+    from qualtran.cirq_interop import CirqQuregT
+    from qualtran.drawing import WireSymbol
+
+
+@frozen
+class SU2RotationGate(GateWithRegisters):
+    r"""Implements an arbitrary SU(2) rotation.
+
+    The rotation is represented by the matrix:
+
+    $$
+        \begin{pmatrix}
+        e^{i(\lambda + \phi)} \cos(\theta) & e^{i\phi} \sin(\theta) \\
+        e^{i\lambda} \sin(\theta) & - \cos(\theta)
+        \end{pmatrix}
+    $$
+
+    References:
+        [Generalized Quantum Signal Processing](https://arxiv.org/abs/2308.01501)
+            Motlagh and Wiebe. (2023). Equation 7.
+    """
+
+    theta: float
+    phi: float
+    lambd: float
```

Comment:
lambda

Reply:
unfortunately as `lambda` is a python keyword, we can't use it here. I could maybe use `lambda_` instead?

Reply:
oh interesting. maybe make a note (comment) about this reason